src/react/providers/room-provider.tsx

```diff
@@ -0,0 +1,79 @@
+import { RoomOptions, RoomOptionsDefaults } from '@ably/chat';
+import React, { ReactNode, useEffect, useMemo } from 'react';
+
+import { RoomContext } from '../contexts/room-context.js';
+import { useChatClient } from '../hooks/use-chat-client.js';
+
+/**
+ * Props for the RoomProvider component.
+ *
+ * Only `id` (the room ID) is required, all other props are optional.
+ */
+export interface RoomProviderProps {
+  /** Required. The id of the room. */
+  id: string;
+
+  /**
+   * Optional. The options to use when creating the room. If none are provided,
+   * the default is {@link RoomOptionsDefaults}.
+   */
+  options?: RoomOptions;
+
+  /**
+   * Optional and defaults to true. Set to false to disable auto-releasing the
+   * room when component unmounts, to support multiple RoomProviders for the
+   * same room.
+   *
+   * If set to false, you must manually release the room using
+   * `chatClient.rooms.release(id)` or have another RoomProvider for the same
+   * room and release set to true.
+   */
+  release?: boolean;
+
+  /**
+   * Optional and defaults to true. Set to false to disable auto-attaching the
+   * room when component mounts and auto-detaching when it unmounts.
+   *
+   * If set to false, you must manually attach and detach the room using
+   * room.attach() and room.detach(). This is useful if you have more providers
+   * for the same room and want to control the attachment manually.
+   */
+  attach?: boolean;
+
+  /** Children nodes. */
+  children: ReactNode;
+}
+
+export const RoomProvider: React.FC<RoomProviderProps> = ({
+  id: roomId,
+  options,
+  release = true,
+  attach = true,
+  children,
+}) => {
+  const client = useChatClient();
+  if (options === undefined) {
```

Comment:
Is it worth setting RoomOptionsDefaults in the props as you've done with release and attach? :)

Reply:
done